teps/0070-tekton-catalog-task-platform-support.md

```diff
@@ -0,0 +1,137 @@
+authors:
+- '@barthy1'
+---
+
+# TEP-0070: Platform support in Tekton catalog
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Catalog side](#catalog-side)
+  - [Hub side](#hub-side)
+  - [Tekton CLI side](#tekton-cli-side)
+- [References (optional)](#references-optional)
+
+## Summary
+
+This TEP proposes to add information about supported operating systems/hardware architectures (platforms) to the Tekton catalog resources and provide ability to the user to filter/search resources from the catalog for specific platform.
+
+## Motivation
+
+Tekton at this moment has releases for 4 hardware architectures(platforms) - `linux/amd64`, `linux/s390x`, `linux/ppc64le` and `linux/arm64`. Tekton catalog contains already great list of resources to run. However there is no information about which resource can be executed on which platform. As a result user can fail running a task on top of not supported hardware architecture.
+
+### Goals
+
+- Define the way to add supported platforms into resource definition in the Tekton catalog.
+- Define the ways to show information about platforms to the user.
+
+### Non-Goals
+
+- Define modifications in Tekton to better support running tasks on different platforms ([TEP-0041](https://github.com/tektoncd/community/pull/310)).
+- Limit/hide ("no-show" or "no-access") catalog resources based on host platform.
+
+### Use Cases (optional)
+
+- Filter the resources in the Hub UI using platforms.
+- Search the resources with `tkn hub` for specific platform(s).
+- Get information about supported platforms for specific resource in the Hub UI/`tkn hub`.
+
+## Requirements
+
+- Platform information in the catalog resource should be optional.
+- Information how to identify/prove supported platform for the resource should be documented in the Tekton catalog.
+- Tekton Hub should automatically get/update information about platforms of the resources.
+- Hub UI should show all platforms available in database to filter.
+
+## Proposal
+
+### Catalog side
+
+Add new `tekton.dev/platforms` annotation to the resource yaml file with list of platforms separated by comma, for example
+
+```
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/tags: image-build
+    tekton.dev/platforms: linux/amd64,linux/s390x
+```
+
+The `tekton.dev/platforms` annotation should be optional. Best practice is to provide information about supported platforms, which should be documented in the catalog README.
+If the annotation is specified, it should not be extended, modified at the processing steps. For instance, if only `linux/ppc64le` value is provided, `linux/amd64` should not be meant as also supported. 
+
+In case if the annotation is not specified, `linux/amd64` platform is meant by default, as this platform is actually default one in cloud/container world.
```

Comment:
to clarify, does this mean that the `tekton.dev/platforms: linux/amd64` annotation will be added by default or is it that it won't be added but we'll understand missing platform annotation to mean that it's `linux/amd64`?

if it's the latter, where do you see us documenting this default behavior? wondering if it may be better to be explicit by requiring at least one platform (and bump all tasks versions)